### ISSUE TITLE
talon-rust log viewer window issues

### ISSUE THREAD

**Issue by knausj85** [talonvoice/talon#555]
Version: 0.3.1-382-g4a86, Branch: rust, OS: macOS-13.2-x86_64-i386-64bit
Version: 0.3.1-382-g4a86, Branch: rust, OS: Windows-10-10.0.19045

- [ ] log viewer can't be closed via command-w on mac, or via alt-f4 on windows

- [x] Log viewer can't be repositioned using the snap commands on mac
2023-02-25 22:51:42 ERROR cb error topic="phrase" cb=SpeechSystem.engine_event
   15:       lib/python3.9/threading.py:937* # cron thread
   14:       lib/python3.9/threading.py:980*
   13:       lib/python3.9/threading.py:917*
   12:                    talon/cron.py:151|
   11:                     talon/vad.py:21 |
   10: talon/scripting/speech_system.py:362|
    9:             talon/engines/w2l.py:722|
    8:      talon/scripting/dispatch.py:134| # 'phrase' main:EngineProxy._redispatch()
    7: talon/scripting/speech_system.py:64 |
    6: ------------------------------------# [stack splice]
    5:      talon/scripting/dispatch.py:134| # 'phrase' main:SpeechSystem.engine_event()
    4: talon/scripting/speech_system.py:415|
    3:      talon/scripting/executor.py:110|
    2:  talon/scripting/talon_script.py:705|
    1:  talon/scripting/talon_script.py:608|
talon.scripting.talon_script.TalonScriptError:
 in script at /Users/knausj/.talon/user/knausj_talon/core/windows_and_tabs/window_management.talon:12:
 > user.snap_window(window_snap_position)
UIErr: window not found

**Comment by knausj85** [comment]
These issues also apply to the fancy new debug window. 

Sometimes, using the snap commands when these windows are focused causes a hard crash on mac

**Comment by lunixbochs** [comment]
snapping and the crash should be fixed on 0.3.1-512, can you confirm?

**Comment by knausj85** [comment]
The crash is fixed, as is the snapping for the most part.

Snapping works until you start getting fancy:

- [x] Snapping can get weird with imgui gaining focus 
- open log viewer
- open command history (or any imgui)
- start chrome (or another app)
- Kill chrome 
at this point, I would expect the log viewer to have focus, but it doesn't.
- focus log viewer 
- issue a snap command (e.g. snap left)
- notice the imgui is focused and reacts to the snap commands

This probably makes more sense as a separate bug?

**Comment by lunixbochs** [comment]
is snapping better now with the focus fixes?

**Comment by knausj85** [comment]
Yes, the snapping's much better. Wrote up a separate issue for more edge-casey stuff.
